windows/src/desktop/kmshell/main/Keyman.System.DownloadUpdate.pas

```diff
@@ -0,0 +1,203 @@
+    begin
+      try
+        http := THttpUploader.Create(nil);
+        try
+          http.Proxy.Server := GetProxySettings.Server;
+          http.Proxy.Port := GetProxySettings.Port;
+          http.Proxy.Username := GetProxySettings.Username;
+          http.Proxy.Password := GetProxySettings.Password;
+          http.Request.Agent := API_UserAgent;
+
+          http.Request.SetURL(url);
+          http.Upload;
+          if http.Response.StatusCode = 200 then
+          begin
+            fs := TFileStream.Create(savepath, fmCreate);
+            try
+              fs.Write(http.Response.PMessageBody^, http.Response.MessageBodyLength);
+            finally
+              fs.Free;
+            end;
+            Result := True;
+          end
+          else // I2742
+          begin
+            // If it fails we set to false but will try the other files
+            Exit(False);
+          end;
+        finally
+          http.Free;
+        end;
+      except
+        on E:EHTTPUploader do
+        begin
+          if (E.ErrorCode = 12007) or (E.ErrorCode = 12029)
+            then LogMessage(S_OnlineUpdate_UnableToContact)
+            else LogMessage(WideFormat(S_OnlineUpdate_UnableToContact_Error, [E.Message]));
+          Result := False;
+        end;
+      end;
+    end;
+
+begin
+  Result := False;
+
+  FDownload.TotalSize := 0;
+  FDownload.TotalDownloads := 0;
+
+  // Keyboard Packages
+  FDownload.StartPosition := 0;
+  for i := 0 to High(Params.Packages) do
+  begin
+    Inc(FDownload.TotalDownloads);
+    Inc(FDownload.TotalSize, Params.Packages[i].DownloadSize);
+    Params.Packages[i].SavePath := SavePath + Params.Packages[i].FileName;
+    if not DownloadFile(Params.Packages[i].DownloadURL, Params.Packages[i].SavePath) then // I2742
+    begin
+      Params.Packages[i].Install := False; // Download failed but install other files
+    end;
+    FDownload.StartPosition := FDownload.StartPosition + Params.Packages[i].DownloadSize;
+  end;
+
+  // Keyman Installer
+  Inc(FDownload.TotalDownloads);
+  Inc(FDownload.TotalSize, Params.InstallSize);
+  if not DownloadFile(Params.InstallURL, SavePath + Params.FileName) then  // I2742
+  begin
+    // TODO-WINDOWS-UPDATES: #10210  convert to error log.
+    LogMessage('DoDownloadUpdates Failed to download' + Params.InstallURL);
+    InstallerDownloaded := False;
+  end
+  else
+  begin
+    InstallerDownloaded := True;
+  end;
+
+  // If installer has downloaded that is success even
+  // if zero packages where downloaded.
+  if InstallerDownloaded then
+    Result := True;
+end;
+
+function TDownloadUpdate.DownloadUpdates: Boolean;
+var
+  DownloadBackGroundSavePath : String;
+  DownloadResult : Boolean;
+  ucr: TUpdateCheckResponse;
+begin
+  DownloadBackGroundSavePath := IncludeTrailingPathDelimiter(TKeymanPaths.KeymanUpdateCachePath);
+  if TUpdateCheckStorage.LoadUpdateCacheData(ucr) then
+  begin
+    DoDownloadUpdates(DownloadBackGroundSavePath, ucr, DownloadResult);
```

Comment:
I'd just use `Result` directly as the parameter here (or better, make `DoDownloadUpdates` into a function:

```
Result := DoDownloadUpdates(DownloadBackGroundSavePath, ucr);
```